.github/workflows/job-unit-tests.yml

```diff
@@ -70,8 +70,8 @@ jobs:
       coveralls_name: ${{ steps.lcov.outputs.name }}
     steps:
       - name: 'Setup deps - apt-get'
-        if: ${{ inputs.cc == 'gcc-9' }}
-        run: sudo apt-get update && sudo apt-get install -y gcc-9 libstdc++-9-dev lcov
+        if: ${{ !env.ACT }}
```

Comment:
This is perhaps the only thing I would consider changing: with this `if` condition, if we were to bump the version number in the CI workflows, but forget to bump it in the Act image Dockerfile, the Act run would be failing since this step is skipped. If we always enable this step, at worst the `apt-get install` would be skipped (packages already installed in the Act image), and at best it would install the newer versions, and ensure the Act run continues. Basically, the Act image installation would only be an optimization (early install). Makes sense?

Reply:
Oh, and also we should have an `os` check for this step, since once we re-enable macOS jobs it will have to be skipped.

Reply:
Indeed. Thank you for the good points.